MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -324,14 +368,69 @@ private void HandleListGuild(GameSession session, PacketReader packet)
             // TODO: update guild listing
         }
 
+        private void HandleSubmitApplication(GameSession session, PacketReader packet)
+        {
+            long guildId = packet.ReadLong();
+
+            Guild guild = GameServer.GuildManager.GetGuildById(guildId);
+            if (guild == null)
+            {
+                return;
+            }
+
+            long guildApplicationId = GuidGenerator.Long();
+
+            session.Send(GuildPacket.SubmitApplicationConfirm(guildApplicationId, guild));
+            guild.BroadcastPacketGuild(GuildPacket.SendApplication(guildApplicationId, session.Player)); // maybe only sent to guild leader/jrs?
+            // TODO: Create a way to store their applicationId with their player information
+        }
+
+        private void HandleApplicationResponse(GameSession session, PacketReader packet)
+        {
+            long guildApplicationId = packet.ReadLong();
+            byte response = packet.ReadByte();
+
+            Guild guild = GameServer.GuildManager.GetGuildById(session.Player.GuildId);
+            if (guild == null)
+            {
+                return;
+            }
+
+            if (response == 0)
+            {
+                //TODO: reject packets
+                return;
+            }
+
+            session.Send(GuildPacket.ApplicationResponseSend(guildApplicationId, response));
+            // TODO: Send 0x06, 0x2F, 0x12, 0x20 to guild/guildLeader
+            // TODO: Send 0x30, 0x5, 0x4B, 0x0, 0x20 to new member
+        }
+
         private void HandleGuildWindowRequest(GameSession session, PacketReader packet)
         {
-            session.Send(GuildPacket.GuildWindow());
+            session.Send(GuildPacket.GuildWindowConfirm());
         }
 
-        private void HandleList(GameSession session, PacketReader packet)
+        private void HandleGuildListRequest(GameSession session, PacketReader packet)
         {
-            session.Send(GuildPacket.List());
+            List<Guild> guildList = GameServer.GuildManager.GetGuildList();
+            guildList = guildList.OrderBy(g => g.Members.Count).ToList();
+            session.Send(GuildPacket.DisplayGuildList(guildList));
+        }
+
+        private void HandleUseBuff(GameSession session, PacketReader packet)
+        {
+            int buffID = packet.ReadInt();
+
+            if (session.Player.Wallet.Meso.Amount < 100000)
```

Comment:
Same thing here (read my other comment) you don't need to access Amount if Modify already does the check for you.